### PR TITLE
Add a counter to filter button and tab

### LINKED ISSUE #482: Filter counter in button and tab
## Problem
The new header simplifies its layout by grouping elements as the viewport goes smaller, but at the same time, it hides the counter label of filters applied when visitors play with the feature.

This element helps users to remember the criteria set to narrow the results, and to convey the filter reset when shifting to a content type that does not match the filter criteria. For this second case, it is crucial to be clear that filters are no longer applied in this new content type.

Due to these two reasons, it becomes necessary to bring back the value and make it consistent in both the filter button and the drawer.

### Related issues
The problem was identified by @zackkrida in WordPress/openverse-frontend#1861

---

### Desktop flow. `lg`
https://user-images.githubusercontent.com/895819/205038179-13eacfd6-f0c4-4248-aee5-05c8a762cf8f.mp4

### Mobile flow. `sm` 
https://user-images.githubusercontent.com/895819/205038139-80d76b7c-b163-4b57-ad61-637127fed7f1.mp4

### Mockups
|Desktop `lg`|Mobile `sm`|
|:---|:---|
|Filter counter (xl)|Filter counter (sm)|

## Description
Now that we can separate the counter from the string, replacing the filter icon with the counter solves the shaking effect of the width change in the header.

The design keeps the `24px` box for both filter icon and the counter across all breakpoints. For the filter button, the resting state has `dark-charcoal-10` as background color to match the search input, whereas the active state has no background.

In the case of the drawer, both tabs have an icon visible all the time. The “Content type" tab shows the active content while “Filters” works as its desktop version to reach more consistency. Since both tabs show an icon, the label’s text style is `label regular` to reduce the layout contrast.

In `lg` breakpoint, the counter has a background just in the icon-size area,  while its active state has a background in the button-size area, as shown below.

Filter counter LG

Since the design proposes to move the counter number into the drawer’s tab, there is no need to show it in the "Clear filters" action. In that vein, we reach consistency with its desktop version by showing the same element.

---

## Feedback
The changes impact the drawer mostly, yet the tweaks make the flow more clear and consistent across breakpoints. What do you think of the idea?

## Design assets
* [Design presentation](https://www.figma.com/proto/vyDzNspOwaNrD1gEaN6wuT/Filter-counter?page-id=26%3A13657&node-id=26%3A13657&viewport=34%2C619%2C0.16&scaling=contain&starting-point-node-id=635%3A16777)
* [Mockups in Figma](https://www.figma.com/file/vyDzNspOwaNrD1gEaN6wuT/Filter-counter?node-id=22%3A13656)
* [Prototype. `xs` breakpoint](https://www.figma.com/proto/vyDzNspOwaNrD1gEaN6wuT/Filter-counter?page-id=626%3A18725&node-id=626%3A20517&viewport=913%2C581%2C0.44&scaling=scale-down&starting-point-node-id=626%3A20517)

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #482 by @panchovm 

## Description

This PR adds the counter to the filters button and tab according to the [Mockups in Figma](https://www.figma.com/file/vyDzNspOwaNrD1gEaN6wuT/Filter-counter?node-id=22%3A13656)

It is the same as https://github.com/WordPress/openverse-frontend/pull/2143, with the gray square on hover (reported by @zackkrida) fixed.

## Testing Instructions

When you select filters, the number of selected filters should appear on the button (desktop) or the tab (mobile).
I've added some Storybook snapshot tests and Playwright VR snapshot tests.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
LGTM

File: frontend/src/components/VHeader/VFilterIconOrCounter.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <VIcon v-if="showIcon" :icon-path="filterIcon" />
+  <p
+    v-else
+    class="flex h-6 w-6 items-center justify-center"
```
Comment (panchovm) [inline frontend/src/components/VHeader/VFilterIconOrCounter.vue:5]: The counter needs to have a corner radius of `2px`. I think the class is `rounded-sm`.

**Comment by panchovm** [comment]
The filter looks great ✨ 

I just noticed two things:
1. The counter needs a corner radius of `2px`. I left the comment above this one.
2. The "clear action" in `VFilterTab` on small breakpoints does not need to show the number. The "Clear filters" label is enough.

| PR | Design
| --- | --- |
|CleanShot 2023-03-13 at 11 45 36@2x|CleanShot 2023-03-13 at 11 53 57@2x|

**Review by zackkrida — APPROVED** [review]
The code and visuals look good to me, but I'd like to make sure someone accessibility-minded can look at this, specifically for the aria label. Previously, the button was labeled "Filters" and its purpose is to toggle the filter sidebar open or closed. Now, it also indicates _how many_ filters are enabled at a given time. 

I wonder if the aria label should be something like "Toggle filters (5 enabled)" instead of the current "5 filters" 🤔

**Comment by obulat** [comment]
> I wonder if the aria label should be something like "Toggle filters (5 enabled)" instead of the current "5 filters" 🤔

I am going to open an issue for better accessibility of the filters button and tab.

**Pull request merged**